iroha_client/src/client.rs

```diff
@@ -198,32 +198,42 @@ impl Client {
         instructions: Vec<Instruction>,
         metadata: UnlimitedMetadata,
     ) -> Result<Hash> {
+        struct EventListenerInitialized;
```

Comment:
I don't think it is good to declare struct inside function. Can we be sure it never will be required outside of the function?

Reply:
Yes, it is used only there for communication between the thread that this function spawns and the functions ordinary code.

Reply:
It shouldn't be used anywhere else, that's why it's local.